### PR TITLE
fix: handle optional IMAAdsManagerDelegate calls

### PR DESCRIPTION
**Pull request by daytime-em**
Swallowing the exceptions is one option. The other is to require `customerAdsManagerDelegate` to extend `NSObject` to be able to use `NSObject respondsToSelector:`. On the one hand, maybe that's reasonable for most callers, on the other it is technically a breaking change and might not be as reasonable as I think it is. 

I think this way might be fine, but I'm open to discussion

### CODE REVIEW

File: Sources/MuxStatsGoogleIMAPlugin/MUXSDKIMAAdsListener.m [hunk 1]
```diff
@@ -291,14 +299,53 @@ - (void)adsManagerDidRequestContentPause:(nonnull IMAAdsManager *)adsManager {
...
     }
 }
 
-/* IMAAdsLoaderDelegate */
+#pragma mark IMAAdsManagerDelegate optional methods
+
+- (void)adsManager:(IMAAdsManager *)adsManager
+adDidProgressToTime:(NSTimeInterval)mediaTime
+         totalTime:(NSTimeInterval)totalTime {
+    [self trySafely:^{
```
Comment (andrewjl-mux) [inline Sources/MuxStatsGoogleIMAPlugin/MUXSDKIMAAdsListener.m:314]: Obj-C has a built in way to do the kind of check that `trySafely` does by [calling `respondsToSelector`](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418583-respondstoselector) which returns a boolean.

Comment (daytime-em) [reply Sources/MuxStatsGoogleIMAPlugin/MUXSDKIMAAdsListener.m:314]: As in the PR text, we need the customer delegates to be instances NSObject. They don't, so there's no way to use respondsToSelector

The main question I have is whether we should accept things as they are and use this approach, or should we require customers' delegates to also inherit from NSObject

Comment (daytime-em) [reply Sources/MuxStatsGoogleIMAPlugin/MUXSDKIMAAdsListener.m:314]: Or even a secret third option that I'm not aware of

Comment (CFKevinRef) [reply Sources/MuxStatsGoogleIMAPlugin/MUXSDKIMAAdsListener.m:314]: Yeah it's typical to have to use `respondsToSelector:` with a delegate that doesn't explicitly inherit from `NSObject`. Cast the delegate to a bare `id` just for the check: `[(id)delegate respondsToSelector:@selector(...])`

**Review by andrewjl-mux — APPROVED** [review]

**Pull request merged**